src/realm/sync/noinst/client_reset_recovery.cpp

```diff
@@ -989,14 +1021,55 @@ void RecoverLocalChangesetsHandler::operator()(const Instruction::Clear& instr)
         ClearResolver(RecoverLocalChangesetsHandler* applier, Instruction::Clear& instr)
             : RecoveryResolver(applier, instr, "Clear")
         {
+            switch (instr.collection_type) {
+                case Instruction::CollectionType::Single:
+                    break;
```

Comment:
I think this is actually REALM_UNREACHABLE ?

Reply:
we could be recovering instructions from a client that did not generate the collection type, so we play safe (same in the InstructionApplier)